src/ray/protobuf/common.proto

```diff
@@ -49,6 +49,30 @@ enum TaskType {
   DRIVER_TASK = 3;
 }
 
+enum LabelMatchOperator {
```

Comment:
I'm thinking of an alternative proto definition that matches the API:

```
message In {
    repeated string values = 1;
}
message Exists {
}
message Operator {
oneof operator {
  In in = 1
  Exists exists = 2
}
}
message NodeLabelSchedulingStrategy {
  map<string, Operator> hard = 1;
  map<...> soft = 2;
}
```

Reply:
1. I think using an array structure would be better for implementing expressions internally, as it would allow for easier expansion in the future for scenarios where there are multiple expressions for the same key, such as {"gpu_type": [Exist(), NotIn("A100")]} without requiring major modifications.

2. This array structure would also result in a simpler hierarchy.